packages/jazz-tools/src/tools/subscribe/CoValueCoreSubscription.ts

```diff
@@ -1,76 +1,196 @@
+      )
+      .then((value) => {
+        if (this.unsubscribed) return;
+
+        if (value !== "unavailable") {
+          // Branch checkout successful, subscribe to it
+          this.subscribe(value);
+        } else {
+          // Branch checkout failed, handle the error
+          this.handleUnavailableBranch();
+        }
+      })
+      .catch((error) => {
+        // Handle unexpected errors during branch checkout
+        console.error(error);
+        this.listener("unavailable");
+      });
+  }
+
+  /**
+   * Handles the case where a branch checkout fails.
+   * Determines whether to retry or report unavailability.
+   */
+  private handleUnavailableBranch(): void {
+    const source = this.node.getCoValue(this.id as any);
+    if (source.isAvailable()) {
+      // This should be impossible - if source is available, branch should be too
+      throw new Error("Branch is unavailable");
+    } else {
+      // Source isn't available either, subscribe to state changes and report unavailability
+      this.subscribeToUnavailableSource();
+      this.listener("unavailable");
     }
   }
 
-  subscribeToState() {
-    const entry = this.node.getCoValue(this.id as any);
+  /**
+   * Loads the CoValue core from the network/storage.
+   * This is the fallback strategy when immediate availability fails.
+   */
+  private loadCoValue(): void {
+    this.node
+      .loadCoValueCore(this.id as RawCoID, undefined, this.skipRetry)
+      .then((value) => {
+        if (this.unsubscribed) return;
+
+        if (value.isAvailable()) {
+          // Loading successful, subscribe to the loaded value
+          this.subscribe(value.getCurrentContent());
+        } else {
+          // Loading failed, subscribe to state changes and report unavailability
+          this.subscribeToUnavailableSource();
+          this.listener("unavailable");
+        }
+      })
+      .catch((error) => {
+        // Handle unexpected errors during loading
+        console.error(error);
+        this.listener("unavailable");
+      });
+  }
+
+  /**
+   * Subscribes to state changes of an unavailable CoValue source.
+   * This allows the subscription to become active when the source becomes available after a first loading attempt.
+   */
+  private subscribeToUnavailableSource(): void {
+    const source = this.node.getCoValue(this.id as any);
+
     const handleStateChange = (
-      core: CoValueCore,
+      _: CoValueCore, // Unused parameter, but required by the subscription API
       unsubFromStateChange: () => void,
     ) => {
       if (this.unsubscribed) {
         unsubFromStateChange();
         return;
       }
 
-      if (core.isAvailable()) {
-        this.subscribe(core.getCurrentContent());
-        unsubFromStateChange();
+      // Source became available, handle it appropriately
+      if (source.isAvailable()) {
+        if (this.branch) {
+          // Branch was requested, attempt checkout again
+          this.handleBranchCheckout();
```

Comment:
In this case, `unsubFromStateChange` is not called. Will this trigger any state changes?